hazelcast/src/test/java/com/hazelcast/map/impl/eviction/PluggableMemoryInfoAccessorTest.java

```diff
@@ -0,0 +1,51 @@
+package com.hazelcast.map.impl.eviction;
+
+import com.hazelcast.config.Config;
+import com.hazelcast.config.MaxSizeConfig;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.IMap;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.annotation.ParallelTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import static com.hazelcast.config.EvictionPolicy.LFU;
+import static com.hazelcast.config.MaxSizeConfig.MaxSizePolicy.FREE_HEAP_PERCENTAGE;
+import static java.lang.System.setProperty;
+import static org.junit.Assert.assertEquals;
+
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelTest.class})
+public class PluggableMemoryInfoAccessorTest extends HazelcastTestSupport {
+
+
+    /**
+     * Used {@link ZeroMemoryInfoAccessor} to evict every put, map should not contain any entry after this test run.
+     */
+    @Test
+    public void testPluggedMemoryInfoAccessorUsed() throws Exception {
+        setProperty("hazelcast.memory.info.accessor.impl", ZeroMemoryInfoAccessor.class.getCanonicalName());
```

Comment:
we should restore the original property after the test is done. 
I believe the Serial runner does this automatically, but I am not sure about the Parallel runner.